src/prerenderContent/prerenderSitemap.ts

```diff
@@ -0,0 +1,65 @@
+import * as Scrivito from 'scrivito'
+import jsontoxml from 'jsontoxml'
+import { storeResult } from './storeResult'
+import { getSiteIds } from './getSiteIds'
+
+export async function prerenderSitemap(
+  targetDir: string,
+  objClassesWhitelist: string[],
+): Promise<void> {
+  console.time('[prerenderSitemap]')
+
+  const pages = await Scrivito.load(() =>
+    prerenderSitemapSearch(objClassesWhitelist).take(),
+  )
+  const sitemapUrls = await Scrivito.load(() => pages.map(pageToSitemapUrl))
+  const content = sitemapUrlsToSitemapXml(sitemapUrls)
+
+  storeResult(targetDir, { filename: '/sitemap.xml', content })
+
+  console.log(
+    `  📦 [prerenderSitemap] Added sitemap.xml with ${sitemapUrls.length} items to ${targetDir}.`,
+  )
+
+  console.timeEnd('[prerenderSitemap]')
+}
+
+function prerenderSitemapSearch(objClassesWhitelist: string[]) {
+  return Scrivito.Obj.onAllSites()
+    .where('_objClass', 'equals', objClassesWhitelist)
+    .and('robotsIndex', 'equals', true)
+    .and('_siteId', 'equals', getSiteIds())
+}
+
+function pageToSitemapUrl(page: Scrivito.Obj) {
+  return {
+    url: {
+      loc: Scrivito.urlFor(page),
+      lastmod: formatDate(page.lastChanged()),
```

Comment:
Strangely the resulting output always renders `<lastmod>2024-07-16</lastmod>` for _all_ pages. Can you investigate why this is the case?

Reply:
This appears to be the actual data. Logged an sorted:
```
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T09:39:31.000Z
2024-07-16T11:40:57.000Z
2024-07-16T11:54:21.000Z
2024-07-16T11:55:22.000Z
2024-07-16T12:15:58.000Z
2024-07-16T12:15:58.000Z
2024-07-16T13:05:49.000Z
2024-07-16T13:05:59.000Z
```